### DISCUSSION TITLE
Anwendung der METS-Attribute LABEL und ORDERLABEL in Kitodo

### DISCUSSION THREAD

**Discussion by andre-hohmann** [kitodo/kitodo-production#6277] [General]
# Einleitung
Das Ziel dieser Diskussion ist die Zusammenführung der vielfältigen Aspekte bezüglich der Anwendung der METS-Attribute `LABEL` und `ORDERLABEL` in Kitodo, um eine nachhaltige Lösung zu ermitteln. 
Die Beschreibung ist stark von den Voraussetzungen und Abläufen in der SLUB Dresden beeinflusst. Es können gerne Ergänzungen in den Kommentaren dieser Diskussion eingetragen werden. 

# 1 Quelle für LABEL und ORDERLABEL
Es bestehen mehrere Issues bezüglich der Quelle der Werte für die METS-Attribute `LABEL` und `ORDERLABEL`. 
- https://github.com/kitodo/kitodo-production/issues/4205
- https://github.com/kitodo/kitodo-production/issues/5016 

Häufig wird auf Kitodo.Production 2 verwiesen, in dem die METS-Attribute fest mit bestimmten MODS-Feldern verbunden sind:
- `LABEL` -> `Haupttitel`
- `ORDERLABEL` -> `Haupttitel (Sortierung)`

Dies hat im Nachhinein betrachtet Vorteile, wie zum Beispiel: 
- Die Werte für die jeweils beiden Felder müssen nur einmal erfasst werden
- Es ist keine Konfiguration notwendig

Dieser Ansatz führt jedoch auch zu Nachteilen, wie zum Beispiel: 
- Beide MODS-Felder müssen immer mit Werten befüllt werden, was beim Strukturieren aufwändig ist
- Wenn für Zeitschriftenbände die `Bandnummer` anstatt des `Haupttitel` erfasst wird, wird in `LABEL` nichts eingetragen
- In anderen Anwendungsbereichen sind gegebenenfalls keine Titel verfügbar, so dass auch in diesen Fällen in `LABEL` nichts eingetragen wird. 

In Kitodo.Production 3 kann dies im Regelsatz durch [den Schlüssel `mets:div`](https://github.com/kitodo/kitodo-production/wiki/Regelsatz-3.x#verwenden-mehrerer-metadatendom%C3%A4nen) bestimmt werden. Dies bietet Flexibilität, führt allerdings zu höherem Erfassungsaufwand. 

# 2 Anwendung Export- und Import-XSLT
In Kitodo.Production 3 kann dies innerhalb eines Vorgangs sehr gut mit dem Export-XSLT gelöst werden. In der SLUB Dresden wird aus `Haupttitel` oder aus `Bandnummer`, jeweils `LABEL` und `ORDERLABEL` abgeleitet und eingetragen. 
Für `ORDERLABEL` werden bestimmte Worte, wie Artikel, ... zu Beginn des Titels entfernt - `Haupttitel (Sortierung)` muss somit nicht immer manuell erfasst werden. Zudem sind die Attribute `LABEL` und `ORDERLABEL` immer vorhanden, auch wenn nur ein Wert erfasst wurde. 

Beispiel SLUB Dresden, in dem in den Ausgaben der Wert aus `Bandnummer` in `LABEL` und `ORDERLABEL` abgeleitet und eingetragen wurde:
- http://digital.slub-dresden.de/id1820646769-19250000
- https://digital.slub-dresden.de/oai/?verb=GetRecord&metadataPrefix=mets&identifier=oai:de:slub-dresden:db:id-1820646769-19250000 

Probleme ergeben sich bei hierarchischen Vorgängen, weil beim Export die Attribute `LABEL` und `ORDERLABEL` vor der Anwendung des Export-XSLT in die `<mets:structMap TYPE="LOGICAL" >` des jeweils übergeordneten Vorgangs eingetragen werden. Der Grund dafür besteht darin, dass für die Ermittlung der Werte eines übergeordneten Vorgangs alle untergeordneten Vorgänge mit dem Export-XSLT bearbeitet werden müssten. Dies führt zu unnötigen hohen Ressourcen-Verbrauch, wenn anstatt eines Vorgangs im schlimmsten Fall bis zu [mehrere hundert Vorgänge - hier Zeitungsausgaben](https://digital.slub-dresden.de/oai?verb=GetRecord&metadataPrefix=mets&identifier=oai:de:slub-dresden:db:id-501434038-1919) intern bearbeitet werden müssten. 

Für Vorgänge (Monografien, Handschriften, mehrbändige Werke, …), deren Metadaten aus anderen Katalogen importiert werden, kann das Import-XSLT angepasst werden, um aus `Haupttitel`, … die Werte für die METS-Attribute zu erzeugen und in die METS-Datei einzutragen. 
- https://github.com/kitodo/kitodo-production/issues/4852
- https://github.com/kitodo/kitodo-production/pull/6143 

Dies ist jedoch nicht möglich, wenn Vorgänge für Zeitschriftenbände oder Zeitungsausgaben in Kitodo.Production angelegt werden. Es soll vermieden werden, dass die Anwender und Anwenderinnen zusätzliche Felder und Werte erfassen müssen, weil das Anlegen der Vorgänge bereits sehr umfangreich ist. 
Unter Umständen wäre sinnvoller, anstatt weiterer Metadaten `LABEL` und `ORDERLABEL` über den Regelsatz zu definieren und manuell zu erfassen, Funktionen zu bestimmen, um die Werte maschinell abzuleiten und einzutragen, wie es derzeit im Export-XSLT möglich ist - nur dass dies zum Beispiel durch Speichern im Metadateneditor erfolgt und nicht beim Export. Im Idealfall kann das XSLT angewendet werden. 

In der SLUB Dresden bestehen deshalb viele untergeordnete Vorgänge ohne `LABEL` und `ORDERLABEL`, was zu negativen Auswirkungen in der Anzeige bestimmter Systeme führt. 

# 3 Anzeige DFG-Viewer, Kitodo.Presentation
In Kitodo.Presentation werden zur Anzeige der Titel der untergeordneten Vorgänge im Strukturbaum Werte aus der Datenbank angewendet, so dass in der SLUB Dresden die zuvor beschriebenen fehlenden `LABEL` und `ORDERLABEL` derzeit nicht auffallen. 
Im DFG-Viewer sind die fehlenden `LABEL` und `ORDERLABEL` in den übergeordneten Vorgängen jedoch sehr rasch erkennbar: 
- https://github.com/kitodo/kitodo-presentation/pull/1315#issuecomment-2323394968 

Im [METS-Anwendungsprofil](https://dfg-viewer.de/fileadmin/groups/dfgviewer/METS-Anwendungsprofil_2.3.1.pdf) wird gefordert, dass auch in Überordnungen die METS-Attribute `LABEL` und `ORDERLABEL` der untergeordneten Vorgänge enthalten sind. Dies spricht dafür, die METS-Attribute `LABEL` und `ORDERLABEL` in allen Vorgängen entsprechend anzuwenden, so dass die exportierte METS-Datei diese Attribute enthält. 

# 4 Anwendung `LABEL` und `ORDERLABEL` in Kitodo.Production
Die Anwendung der Attribute `LABEL` und `ORDERLABEL` kann auch in Kitodo.Production angepasst werden, um die Anzeige und Performance im Metadateneditor zu verbessern. Es müssen jedoch einige Herausforderungen berücksichtigt werden, die in folgender Diskussion beschrieben sind: 
- https://github.com/kitodo/kitodo-production/discussions/6184 

# 5 Altdaten
Manuelle Korrekturen der Altdaten sind in der SLUB Dresden nicht möglich. Für Altdaten wäre es somit sehr hilfreich, wenn mit den Kitodo-Skripten die Erfassung der fehlenden METS-Attributen `LABEL` und `ORDERLABEL` unterstützt würde. 
- https://github.com/kitodo/kitodo-production/issues/6216 

Mit dieser Anpassung könnten mit relativ überschaubaren Aufwand die Werte aus den MODS-Feldern `Haupttitel` oder `Bandnummer` in `LABEL` und `ORDERLABEL` kopiert werden. 

Falls eine Funktion bereitgestellt würde, die es erlaubt, `LABEL` und `ORDERLABEL` nach bestimmten Regeln beim Speichern des Metadateneditors zu erstellen, ist die beschriebene Anpassung der Kitodo-Skripte unter Umständen nicht notwendig. 
Ein Kitodo-Skript zum Speichern der METS-Dateien mehrerer Vorgänge wäre sinnvoller.

**Comment by andre-hohmann** [comment]
Mir ist noch eingefallen, dass Werte für `ORDERLABEL` der Jahres-Vorgänge (1860, 1861, 1863, ....) in das MODS-Element `Haupttitel (Sortierung)` eingetragen werden mussten. 
Dies führte beim Export dazu, dass wegen dieser Metadaten viele `<dmdSec>` in die übergeordneten Vorgänge der Zeitungen erstellt wurden, die die `Haupttitel (Sortierung)` der Jahres-Vorgänge enthielten. 

Ich empfinde es als Vorteil, dass dies durch die Trennung METS-Attribut und MODS-Elemente nicht mehr auftritt. 

**Nachtrag**:
Vermutlich würde dieses Verhalten in Kitodo.Production 3 unter anderem wegen der Auftrennung der Hierarchien nicht mehr in dieser Form auftreten. Trotzdem zeigt dieses Beispiel, dass es Probleme verursachen kann, wenn die METS-Attribute und MODS-Elemente immer miteinander angewendet werden müssen.

**Comment by BartChris** [comment]
Danke für die sehr instruktive Zusammenstellung. Ein möglicher Vorschlag wäre, das Ruleset um weitere funktionelle Keys/Attribute zu ergänzen.

```xml
<division id="periodical" processTitle="'MY_ID_'+zdb_id+'_MY_ID'" withWorkflow="false"
            use="createChildrenFromParent">
            <label>Periodical</label>
            <label lang="de">Zeitschrift</label>
</division>
```

Es ist bereits jetzt möglich, den Vorgangstitel aus anderen Feldern (hier: zdb_id) mit Freitext kombiniert zusammenzusetzen. Man könnte dies für LABEL und ORDERLABEL analog machen:

```xml
<division id="periodical_volume" processTitle="'MY_ID_'+zdb_id+'_MY_ID'" LABEL="TitleDocMain" ORDERLABEL="TitleDocMain" withWorkflow="false"
            use="createChildrenFromParent">
            <label>Periodical</label>
            <label lang="de">Zeitschrift</label>
</division>
```

Dann könnte bereits bei der Anlage des Vorgangs ein entsprechendes LABEL und ORDERLABEL besetzt werden, das nachträglich natürlich änderbar ist. Bei Vorgängen, die eigentlich nur "LABEL" und keinen Haupttitel benötigen, könnte man bei Anlage des Vorgangs LABEL befüllen und Haupttitel als Feld optional befüllbar machen.

**Reply by andre-hohmann** [reply]
Vielen Dank - das klingt nach einem guten Lösungsansatz!

Für Zeitschriftenbände der SLUB Dresden würde `CurrentNo` anstatt `TitleDocMain` angewendet werden: 

```xml
<division id="PeriodicalVolume" processTitle="'TSL_ATS+'_'+CatalogIDDigital'" LABEL="CurrentNo" ORDERLABEL="CurrentNo">
    <label>Periodical Volume</label>
    <label lang="de">Zeitschriften-Band</label>
</division>
```

Für Zeitungsausgaben der SLUB Dresden würde vermutlich `TitleDocMain` angewendet werden, um die Ausgabenbezeichnungen einzutragen. Wenn keine Ausgabebezeichnung vorhanden ist, bleiben `LABEL`, `ORDERLABEL` leer. 

```xml
<division id="PeriodicalIssue" processTitle="+'-'+#YEAR+#MONTH+#DAY+#is+'_'+#issu" LABEL="TitleDocMain" ORDERLABEL="TitleDocMain">
    <label>Issue (Newspaper Issue)</label>
    <label lang="de">Heft/Ausgabe (Zeitungs-Ausgabe)</label>
</division>
```

In beiden Fällen sind keine Anpassungen notwendig, wie zum Beispiel des Entfernen von nicht-sortierbaren Wörtern (Artikel, ...). Dies wird beim Import- und Export-XSLT durchgeführt. 
Es scheint auch zu keinen Konflikten zu führen, weil für einen Dokumenttyp unterschiedliche Metadaten die Quelle für ein METS-Attribut sein könnten. 

Vermutlich könnte das in der SLUB Dresden angewendet werden. Es bestehen dann jedoch drei Arten zur Erstellung der METS-Attribute `LABEL`, `ORDERLABEL`: 
1. Erstellung durch Import-XSLT
2. Erstellung durch Regelsatz-Regeln
3. Erstellung durch Export-XSLT

Es schient jedoch beherrschbar und es wird vermutlich dadurch mit relativ geringem Einsatz ein hoher Nutzen erlangt. 
Es wäre jedoch elegant, wenn ein einheitliches Verfahren angewendet werden könnte. Ich werde das noch genauer prüfen.

**Comment by andre-hohmann** [comment]
Ergebnis der Abstimmung über das weitere Vorgehen am 2025-09-29 von @andre-hohmann und @BartChris 

# Anpassung Export
Vergleiche: 
- #6277
## Ablauf

1. Export des untergeordneten Vorgangs
2. Auswertung der exportierten METS-Datei 
    -  <mets:structMap TYPE="LOGICAL" >
3. Extraktion aus <mets:structMap TYPE="LOGICAL" >: 
    - LABEL
    - ORDERLABEL
4. Eintragen in interne METS-Datei (meta.xml)
    - LABEL
    - ORDERLABEL
5. Speichern der internen Datei und Erzeugen neuer Version. 
6. Export des übergeordneten Vorgangs
    - inklusive Auswertung LABEL und ORDERLABEL der untergeordneten Vorgänge

## Vorteile
- Die Logik zur Erstellung der Werte von LABEL und ORDERLABEL wird an einer Stelle verwaltet. 
- Die Anpassung des XSLT ist sehr flexibel. 
- Auf eine redundante Erfassung der Werte kann verzichtet werden. 

## Kritische Punkte
- Anpassung der Export-Reihenfolge (serieller und paralleler Export)


# Anpassung Metadateneditor
Vergleiche: 
- #6184

## Konzept

Anwendung von use="displaySummary", um im Metadateneditor weitere Metadaten-Informationen aus den Kind-Vorgängen im Eltern-Vorgang anzuzeigen. 
Vergleiche: #4736
- Abhängig von der Performance
  - Vergleiche: 
    - #5934
  - Der TYP des untergeordneten Vorgangs wird in Version 3.9 aus der XML-Datei ausgelesen; bis einschließlich 3.8 wurde der TYP dem Index entnommen
- 2.Schritt: Prüfung und Entscheidung, ob use="displaySummary" im Index gespeichert wird

## Vorteile
- Nachnutzung bestehender Logik.

**Reply by henning-gerhardt** [reply]
Zu em Punkt "Anpassung im Export": Wenn dies so umgesetzt wäre, sind dann die folgenden Szenario so richtig bei den Auswirkungen im Dateisystem gedacht? 
1. Es werden 365 einzelne Tagesausgaben einer Zeitung im Laufe eines Tages nacheinander exportiert und sequentiell über Taskmanager exportiert, dann wird 365 Mal die Metadaten-Datei der Jahresebene, in der die einzelnen Tagesausgaben referenziert sind,  in der gleichen Anzahl manipuliert? 
2. Einige Zeit später, nachdem die 365 Tagesausgaben exportiert und präsentiert sind, fällt auf, dass ein Metadatum in den Tagesausgaben falsch ist bzw. korrigiert werden muss. Die Korrektur erfolgt über KitodoScript Aufrufe und danach werden diese Vorgänge über den Taskmanager einzeln und sequentiell neu exportiert: Wird dann die Metadaten-Datei der  Jahresebene 365 Mal erneut bearbeitet, weil die Tagesausgaben exportiert werden?

**Reply by BartChris** [reply]
Ad 1. 
Bereits beim aktuellen Export ist es so, dass ein Export von 365 Tagesausgaben auch 365 mal die Export-Metadaten-Datei der Jahresebene erzeugt, mit dem jeweils zu diesem Zeitpunkt gültigen Kind-Verweisen. 
Die vorgesehene Logik würde nach dem Export der einzelnen Tagesausgabe zusätzlich nur die interne METS-Datei derselben Tagesausgabe manipulieren, nicht aber die interne METS-Datei der Jahresebene.  Ich versuche das gleich mal probehaft im Code umzusetzen um diese Annahmen zu überprüfen.

Ad 2.
Da die interne METS-Datei der Jahresausgabe nicht manipuliert wird, würde auch in diesem Fall jeweils nur die interne METS-Datei des jeweiligen Tages-Vorgangs, der exportiert wird, manipuliert werden

**Reply by henning-gerhardt** [reply]
Dann hatte ich es falsch verstanden, welche Datei geändert wird. 

Die Frage, die ich mir stelle: muss dies wirklich so sein oder kann man nicht eine bessere Alternative finden als bei jedem Export die interne Metadaten-Datei neu zu schreiben und somit auch bei jedem Export eine neue Version zu erzeugen? Wenn man dies "geschickt" macht, dann exportiert man so lange bis die lokale Datei-Rotation, die beim Erzeugen der neuen Version zum Einsatz kommt, bis alle Versionen "gleich" sind und vorherige Versionen maximal noch im systemweiten Backup existieren. Dies sollte man zumindest mit bedenken.

Wäre es vielleicht nicht besser, wenn man bei der Anwendung des Export-XSLT nicht Zugriff auf das gesamte exportierte Ergebnis hätte und nicht nur einen Teil, wie es jetzt wohl der Fall ist?

**Reply by BartChris** [reply]
Hier wäre eine Option vor dem Neu-Schreiben der METS-Datei zu überprüfen, ob es eine wirkliche Änderung an LABEL oder ORDERLABEL gibt. Wenn die beiden vor und nach dem Update identisch blieben = kein Update, keine neue Version.

**Reply by BartChris** [reply]
Ich habe hier einen experimentellen Branch erstellt:
https://github.com/BartChris/kitodo-production/tree/add-export-label-update

Diff: https://github.com/kitodo/kitodo-production/compare/main...BartChris:kitodo-production:add-export-label-update?expand=1

Ich vermute, die Logik funktioniert nur dann robust, wenn die Eltern nach den Kindern exportiert werden. (Das ist  sichergestellt, wenn wir taskManager.autoRunLimit=1 setzen.) Dies ist jedoch bei Verwendung mehrerer Threads nicht garantiert. Allerdings wäre das Fehlverhalten bei mehreren Threads darauf beschränkt, dass die LABELs beim ersten Export im Elternobjekt nicht zwingend vorhanden sind.
Ich denke, das allgemeine Ziel sollte hier sein, dass die Elternprozesse erst am Ende exportiert und nicht mehrfach unnötig exportiert werden. Siehe auch https://github.com/kitodo/kitodo-production/discussions/6633 .

**Reply by M3ssman** [reply]
In diesem Zusammenhang möchte ich den Standpunkt einer Einrichtung einbringen, die viele Jahre mit einem anderen System digitalisiert hat, ehe sie um 2020 zu Kitodo.Production wechselte und seit 2-3 Jahren auch die 3er Version von Kitodo.Production nutzt.

Die Sichtweise beschränkt sich nicht auf Zeitungen/Zeitschriften.

In den Retrodigitalisierungs-Workflows der ULB Sachsen-Anhalt werden Digitalisate (vor allem mit VD-Bezug, mit Ausnahme von Zeitungsausgaben) so gut wie immer im logischen Bereich in die Tiefe strukturiert. Dieses Vorgehen ist mE in Einrichtungen, die Kitodo.Production einsetzen, nicht allgemein verbreitet. 

Vor diesem Hintergrund (und explizit nicht auf Zeitungsausgaben eingegrenzt): Was war der Grund, um mit Kitodo.Production 3 für jedes Element der logischen Struktur eine eingene Metadaten-Sektion einzuführen? Das hat z.B. bei uns den Effekt - weil immer noch stark in die Tiefe strukturiert wird - dass bei umfangreichen Strukturen hunderte zusätzliche `dmdSec` angelegt werden, die nur Titelelemente enthalten. Die müssen dann zusätzlich beim Export bzw. in nachgelagerten Workflows in die `LABEL` der logischen Container kopiert werden, u.A. weil wir das für die PDF-Outlines benötigen. Was mich dabei verwirrt: wieso wurde das so festgelegt, obwohl der DFG-Viewer (und die DDB) diese Label in der logischen Struktur erwarten?

Für unsere Workflows ist das Verhalten von Kitodo.Production 3 nachteilig, weil Zusatzarbeiten für das Bauen einer vollständigen logischen Struktur erforderlich sind, die bei Kitodo.Production 2 noch nicht notwendig waren. @andre-hohmann In #6184 wird erwähnt, dass das Verhalten von Kitodo.Production 2 gewünscht wird, aber die Nachteile dieser Lösung nicht mehr in Erinnerung seien. Frage: was waren die Nachteile?

**Reply by andre-hohmann** [reply]
Es wird nicht für jedes Element in der logischen Struktur eine `<dmdSec>` erstellt, sondern nur, wenn ein im Regelsatz entsprechendes konfiguriertes Metadatum in dem Strukturelement enthalten ist. 

Wenn einem Strukturelement nur `LABEL` zugewiesen wird - durch Anwendung des Attributs `mets:div` im Regelsatz entsprechend [Verwenden mehrerer Metadatendomänen](https://github.com/kitodo/kitodo-production/wiki/Regelsatz-3.x#verwenden-mehrerer-metadatendom%C3%A4nen), wird meines Wissens keine `<dmdSec>` erstellt. 

In Kitodo.Production 2 war es meines Wissens notwendig, die `<dmdSec>` zu erstellen, um daraus die `LABEL` für die logische Struktur zu erstellen. 
Aus diesem Grund wurde die Verbindung aufgehoben: 
- https://github.com/kitodo/kitodo-production/issues/4205#issuecomment-784164621

**Reply by M3ssman** [reply]
@andre-hohmann Vielen Dank für die Erörterung! 

Es ist nicht auszuschließen, dass unsere Regelsätze für Kitodo 3 nicht optimal sind. 
Aktuell ist es so, dass je Strukturelement komplett eigenständige METS/MODS-Sektionen erzeugt werden, in denen nur der Titel steht. Und dann muss beim DMS-Export dafür gesorgt werden, dass diese Information in die logische Struktur kopiert wird. Das bläht die Metadaten auf, wie z.B. in diesem VD17-Werk: [Sacrarum Antiquitatum De Precibus Veterum Christianorum Libellus](https://opendata2.uni-halle.de/oai/dd?verb=GetRecord&metadataPrefix=mets&identifier=oai:opendata2.uni-halle.de:1516514412012/297523)

Für Kitodo 2 waren keine zusätzlichen `dmdSec` erforderlich, um Label für Strukturelemente zu erzeugen. Allerdings war damals viel Logik durch die Zusatzkomponente [kitodo UGH](https://github.com/kitodo/kitodo-ugh) abgedeckt, welche häufig ein schwierig nachzuvollziehendes Verhalten an den Tag legte (z.B. mit sich selbst duplizierenden Metdatenfeldern). Ein Ziel bei Kitodo 3 war es IMHO, diese MET/MODS Komponente zu ersetzen.

**Reply by andre-hohmann** [reply]
In der SLUB benötigen wir die MODS-Elemente für die Strukturelemente, weil in Kitodo.Presentation die MODS-Elemente für die Indexierung ausgewertet werden und auch die Kapitel, ... suchbar und auffindbar sein sollen. 
Teilweise werden auch Personendaten, Ausgabennummern oder andere Informationen erfasst. 
Daher ist es für uns sinnvoll, die MODS-Metadaten zu erfassen und daraus im Export-XSLT das LABEL zu erzeugen. 

Mir war nicht bekannt, dass in Kitodo.Production 2 das LABEL für Strukturelemente ohne `dmdSec` erzeugt werden konnten.

**Reply by M3ssman** [reply]
@andre-hohmann Ich hab mal ein paar Vorgänge geprüft, die 2021 mit Kitodo 2 digitalisiert wurden (z.B. [Geheimniß des Gnaden-Bund, Zeyter Teil](https://opendata.uni-halle.de/oai/dd?verb=GetRecord&metadataPrefix=mets&identifier=oai:opendata.uni-halle.de:1981185920/36475) ), da war es doch schon so, dass für Strukturelemente zusätzliche dmdSecs erstellt wurden - mein Fehler.

**Comment by andre-hohmann** [comment]
Mit folgendem Pull Request werden die Attribute `LABEL` und `ORDERLABEL` in `<mets:structMap TYPE="LOGICAL">` der internen `meta.xml` während des Exports eingetragen, wenn sie von dem bisherigen Inhalt abweichen. 
- #6705 

Dies führt dazu, dass auch in den übergeordneten Vorgängen die Attribute `LABEL` und `ORDERLABEL` der untergeordneten eingetragen werden. 
Die Logik des Export XSLT wird somit in allen Fällen angewendet und muss nur einmal gepflegt werden. 

Dadurch sollten folgende Aspekte gelöst sein: 
- Erstellung der Attribute `LABEL` und `ORDERLABEL` während des Imports durch das XSLT und die dauerhafte Verwaltung in der internen `meta.xml`
  - https://github.com/kitodo/kitodo-production/issues/4852#issuecomment-1173364799
- Korrektur der bestehenden Vorgänge

Ich schließe die Diskussion, weil
1. die wichtigsten Aspekte gelöst sind
2. die Diskussion durch weitere Kommentare unübersichtlich wird und statt dessen eine neue Diskussion mit konkreten Anliegen eröffnet werden sollte

**Comment by andre-hohmann** [comment]
Informationen zur Anwendung oder Konfiguration bitte im Wiki-Artikel eintragen: 
- [Erstellung der METS‐Attribute LABEL und ORDERLABEL in Kitodo](https://github.com/kitodo/kitodo-production/wiki/Erstellung-der-METS%E2%80%90Attribute-LABEL-und-ORDERLABEL-in-Kitodo)
